src/refspec.c

```diff
@@ -160,6 +160,30 @@ void git_refspec__free(git_refspec *refspec)
 	memset(refspec, 0x0, sizeof(git_refspec));
 }
 
+int git_refspec_parse(git_refspec **refspec, const char *input, int is_fetch)
```

Comment:
Minor nit that doesn't need addressing: we usually name out-parameters `out` or similar. In case you rename it to out, you could also just rename `new_refspec` to `refspec`.